### PR TITLE
Enable keyboard navigation and deletion of chosen suggestions

### PR DESCRIPTION
**Pull request by josephschmitt**

### CODE REVIEW

File: src/angularComponent/ngcOmniboxController.js [hunk 1]
```diff
@@ -205,6 +205,22 @@ export default class NgcOmniboxController {
     return this._highlightedIndex;
   }
 
+  set fieldElement(el) {
```
Comment (onlywei) [inline src/angularComponent/ngcOmniboxController.js]: What's the point of using `get`/`set` for this `fieldElement` property?

Comment (josephschmitt) [inline src/angularComponent/ngcOmniboxController.js]: Check out the latest commit. I add an event listener.

File: examples/category-results-materialize.html [hunk 2]
```diff
@@ -4,51 +4,74 @@
...
-        <ngc-omnibox-choices class="chips">
-          <div class="chip">
-            {{selection.make_display}}
-            <i class="close material-icons" ng-click="omnibox.unchoose(selection)">
-              close
-            </i>
-          </div>
-        </ngc-omnibox-choices>
+          <ngc-omnibox-field class="input" type="text" tabindex="0"
+              ng-focus="omnibox.fieldHasFocus = true" ng-blur="omnibox.fieldHasFocus = false">
```
Comment (onlywei) [inline examples/category-results-materialize.html]: We need the app-maker to do this `ng-focus` & `ng-blur` manually?? Why can't we do it internally?

File: spec/tests/angularComponent/ngcOmniboxControllerSpec.js [hunk 3]
```diff
@@ -4,9 +4,14 @@ describe('ngcOmnibox.angularComponent.ngcOmniboxController', () => {
   let omniboxController;
 
   beforeEach(() => {
-    const fakeEl = {removeAttribute: () => {}, setAttribute: () => {}};
-
-    omniboxController = new NgcOmniboxController();
+    const fakeEl = {
```
Comment (onlywei) [inline spec/tests/angularComponent/ngcOmniboxControllerSpec.js]: not `jasmine.createSpyObj()`?

File: src/angularComponent/ngcOmniboxChoicesDirective.js [hunk 4]
```diff
@@ -5,13 +5,17 @@ export default function ngcOmniboxChoicesDirective() {
       omnibox: '^^ngcOmnibox'
     },
     scope: true,
-    controller: () => {},
+    controller() {},
+    controllerAs: 'choices',
     compile(tElement) {
       const element = tElement[0];
-      const tokens = element.firstElementChild;
+      const token = element.firstElementChild;
```
Comment (onlywei) [inline src/angularComponent/ngcOmniboxChoicesDirective.js]: YES!! This confused me every time I read this code before.

File: src/coreComponent/keyboard.js [hunk 5]
```diff
@@ -195,8 +194,7 @@ export function isHorizontalMovementKey(keyCode) {
   switch (keyCode) {
     case KEY.LEFT:
     case KEY.RIGHT:
-    case KEY.BACKSPACE:
-    case KEY.DELETE:
+    case KEY.TAB:
```
Comment (onlywei) [inline src/coreComponent/keyboard.js]: does tab really move the caret horizontally?

File: examples/category-results-materialize.html (hunk 2 above)
Comment (josephschmitt) [inline examples/category-results-materialize.html]: I re-did this in a later commit.

File: spec/tests/angularComponent/ngcOmniboxControllerSpec.js (hunk 3 above)
Comment (josephschmitt) [inline spec/tests/angularComponent/ngcOmniboxControllerSpec.js]: Need to write new tests for all the new stuff I did still.

File: src/angularComponent/ngcOmniboxChoicesDirective.js (hunk 4 above)
Comment (josephschmitt) [inline src/angularComponent/ngcOmniboxChoicesDirective.js]: Yea, not sure what I was thinking before.

File: src/coreComponent/keyboard.js (hunk 5 above)
Comment (josephschmitt) [inline src/coreComponent/keyboard.js]: It moves focus horizontally, but I see your point. I'll kill TAB.

File: examples/category-results-materialize.html (hunk 2 above)
Comment (onlywei) [inline examples/category-results-materialize.html]: Wait, so `fieldHasFocus` is a custom property that the app-maker is attaching to the omnibox controller here? That doesn't really seem like something that we should recommend.

Comment (josephschmitt) [inline examples/category-results-materialize.html]: Oh, right sorry, that's supposed to be `demoCtrl` not `omnibox`. Done.

**Pull request merged**